### PR TITLE
Small libuserhooks refactor. Make it a proper singleton.

### PR DESCRIPTION
**Pull request by kscieslinski**
This is just a very small refactor of libuserhook. This pr. should fix memory leak issues observed in #1060

### CODE REVIEW

File: src/libusermode/uh-private.hpp [hunk 1]
```diff
@@ -165,20 +165,21 @@ struct copy_on_write_result_t : public call_result_t
...
 
     std::vector<usermode_cb_registration> plugins;
     // map dtb -> list of hooked dlls
     std::map<addr_t, std::vector<dll_t>> loaded_dlls;
 
-    userhook(drakvuf_t drakvuf) : pluginex(drakvuf, OUTPUT_DEFAULT), initialized(0)
-    {
-        drakvuf_get_kernel_struct_members_array_rva(drakvuf, offset_names, __OFFSET_MAX, offsets.data());
-    }
+    userhook(drakvuf_t drakvuf);
```
Comment (BonusPlay) [inline src/libusermode/uh-private.hpp:174]: The big brain strat is to declare ctor private, so you are forced to use `get_instance` method.

Comment (kscieslinski) [reply src/libusermode/uh-private.hpp:174]: Thanks, nice advice.

**Review by BonusPlay — APPROVED** [review]

**Comment by icedevml** [comment]
@drakvuf-jenkins This is OK to test.

**Review by icedevml — APPROVED** [review]

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
Hm, got an error `free(): invalid pointer` on jenkins for the last run.

**Comment by icedevml** [comment]
I've also seen something like `GSlice free: invalid pointer` in the manual test. I could provide more details later.

**Comment by kscieslinski** [comment]
> I've also seen something like `GSlice free: invalid pointer` in the manual test. I could provide more details later.

I've found the source of the issue (~userhook::userhook()). I will try to resolve it today.

**Pull request merged**